ursh/indico_ursh/util.py

```diff
@@ -0,0 +1,34 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2018 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+from werkzeug.exceptions import BadRequest
+
+import requests
+
+
+def request_short_url(original_url):
+    from indico_ursh.plugin import UrshPlugin
+    api_key = UrshPlugin.settings.get('api_key')
+    api_host = UrshPlugin.settings.get('api_host')
+    if not api_key or not api_host:
+        raise BadRequest('API key/host missing - have you set the plugin settings?')
```

Comment:
I'd raise a `ServiceUnavailable` error here; it's not related to the client sending the request doing something wrong. I'd also remove the part after the dash, since it's not useful to non-admins. Actually, I would change the plugin signal handlers so they only add the url shortener link if key and host are set.